bp_common/src/include/bp_common_me_if.vh

```diff
@@ -183,105 +181,123 @@ typedef enum bit [1:0]
-{
-  e_VI_I                     = 2'b00
-  ,e_VI_V                    = 2'b10
-} bp_cce_coh_vi_e;
-
-typedef union packed 
+typedef enum bit [2:0] 
 {
-  bp_cce_coh_vi_e            vi;
-  bp_cce_coh_mesi_e          mesi;
-}  bp_cce_coh_u;
+  e_COH_I                   = 3'b000 // Invalid
+  ,e_COH_S                  = 3'b001 // Shared - clean, not owned, shared (not exclusive)
+  ,e_COH_E                  = 3'b010 // Exclusive - clean, owned, not shared (exclusive)
+  ,e_COH_F                  = 3'b011 // Forward - clean, owned, shared (not exclusive)
+  //                          3'b100 - potentially dirty, not owned, not shared (exclusive)
+  //                          3'b101 - potentially dirty, not owned, shared (not exclusive)
+  ,e_COH_M                  = 3'b110 // Modified - potentially dirty, owned, not shared (exclusive)
+  ,e_COH_O                  = 3'b111 // Owned - potentially dirty, owned, shared (not exclusive)
+} bp_coh_states_e;
+
+`define bp_coh_shared_bit 0
+`define bp_coh_owned_bit 1
+`define bp_coh_dirty_bit 2
+
+`define bp_coh_bits $bits(bp_coh_states_e)
+
+// LCE Command Network Message Types
+typedef enum logic [1:0] {
+  e_lce_cmd_cmd            // command
+  ,e_lce_cmd_data          // cache block data to LCE, i.e., cache block fill
+  ,e_lce_cmd_uc_data       // unached data to LCE, i.e, up to 64-bits data
+} bp_lce_cmd_type_e;
 
-`define bp_cce_coh_bits $bits(bp_cce_coh_u)
+`define bp_lce_cmd_type_width $bits(bp_lce_cmd_type_e)
 
-/*
- * bp_cce_lce_cmd_s defines a command sent by a CCE to and LCE
- * dst_id is the LCE receiving the command
- * src_id is the CCE sending the command
- * msg_type is the command
- * addr specifies the memory address associated with the command
- * way_id is the way within the set that addr maps to in the LCE that should be used for the command
- * state specifies the Coherence State to be used for invalidate, set_tag, and set_tag_wakeup
- * target is the LCE that will receive a transfer for a transfer command
- * target_way_id is the way within the target LCE's set (computed from addr) to fill the data in to
+/**
+ *  bp_lce_cmd_cmd_s defines a command sent by a CCE to and LCE
+ *
+ *  src_id:        is the CCE sending the command
+ *  msg_type:      is the command
+ *  addr:          specifies the memory address associated with the command
+ *  way_id:        is the way within the set that addr maps to in the LCE that should be used for the command
+ *  state:         specifies the Coherence State to be used for invalidate, set_tag, and set_tag_wakeup
+ *  target:        is the LCE that will receive a transfer for a transfer command
+ *  target_way_id: is the way within the target LCE's set (computed from addr) to fill the data in to
  */
-`define declare_bp_cce_lce_cmd_s(num_cce_mp, num_lce_mp, addr_width_mp, lce_assoc_mp) \
+`define declare_bp_lce_cmd_cmd_s(num_cce_mp, num_lce_mp, addr_width_mp, lce_assoc_mp) \
   typedef struct packed                                         \
```

Comment:
Don't we still need dst_id to determine which LCE to send the command to?

Reply:
yeah, it is in the bp_lce_cmd_s since it is needed for both regular and data commands.